utils/pckgen

```diff
@@ -0,0 +1,559 @@
+        local DEP_NAME="DEPENDENCIES_${package##*_}[@]"
+        DEPENDENCIES+=(${!DEP_NAME})
+    done
+    for DEP in ${DEPENDENCIES[@]}; do
+        which $DEP &> /dev/null
+        if [ $? -ne 0 ]; then
+            local FAILED_DEPS+="$DEP "
+        fi
+    done
+    if [ "$FAILED_DEPS" ]; then
+        error "Dependencies not installed. You need to provide these programs first: $FAILED_DEPS"
+    fi
+}
+
+create_output_dir() {
+    mkdir -p "${OUTPUT_DIR:=$DEFAULT_OUTPUT_DIR}"
+    if [ ! -d "$OUTPUT_DIR" ]; then
+        error "Couldn't create output directory '$OUTPUT_DIR'"
+    fi
+}
+
+cas_version() {
+    mkdir -p "$TEMP_DIR"
+    cd "$TEMP_DIR"
+    sh "$CAS_VERSION_GEN" > /dev/null
+    if [ ! -f "$CAS_VERSION_GEN_FILE" ]; then
+        error "Couldn't create CAS_VERSION file"
+    fi
+    while read -r line; do
+        case "$line" in
+            CAS_VERSION_MAIN*) CAS_VERSION_MAIN="${line##*=}";;
+            CAS_VERSION_MAJOR*) CAS_VERSION_MAJOR="${line##*=}";;
+            CAS_VERSION_MINOR*) CAS_VERSION_MINOR="${line##*=}";;
+        esac
+    done < "$CAS_VERSION_GEN_FILE"
+    rm "$CAS_VERSION_GEN_FILE"
+    cd - > /dev/null
+
+    VERSION_SHORT="${CAS_VERSION_MAIN}.${CAS_VERSION_MAJOR}"
+    if [ $CAS_VERSION_MINOR -ne 0 ]; then
+        VERSION_SHORT+=".${CAS_VERSION_MINOR}"
+    fi
+    if [[ "$VERSION_SHORT" < "$SUPPORTED_FROM_VERSION" ]]; then
+        echo "Sorry... this version of $NAME ($VERSION_SHORT) is not supported"\
+             "by $THIS. Use $NAME >= $SUPPORTED_FROM_VERSION"
+        exit 1
+    fi
+
+    if [ -d "$SOURCES_DIR/.git" ]; then
+        CAS_VERSION_BUILD=$(git -C "$SOURCES_DIR" log --merges --oneline | wc -l)
+        while [ $(echo -n $CAS_VERSION_BUILD | wc -m) -lt 4 ]; do
+            CAS_VERSION_BUILD="0${CAS_VERSION_BUILD}"
+        done
+        LAST_COMMIT_HASH=$(git -C "$SOURCES_DIR" log --pretty=format:%H | head -n 1)
+        LAST_ABBR_COMMIT_HASH=$(git -C "$SOURCES_DIR" log --pretty=format:%h | head -n 1)
+        LAST_COMMIT_DATE=$(git -C "$SOURCES_DIR" log --pretty=format:%cI | head -n 1)
+        LAST_COMMIT_TIMESTAMP=$(git -C "$SOURCES_DIR" log --pretty=format:%ct | head -n 1)
+    elif [ -f "$VERSION_FILE" ]; then
+        source "$VERSION_FILE" &> /dev/null
+    else
+        error "Couldn't obtain CAS version - no git tree nor version file within given sources"
+    fi
+
+    VERSION_LONG="${CAS_VERSION_MAIN}.${CAS_VERSION_MAJOR}.${CAS_VERSION_MINOR}.${CAS_VERSION_BUILD}"
+
+    if [ $MASTER_PACKAGE ]; then
+        VERSION="$VERSION_LONG"
+        RELEASE="master"
+    elif [ $RELEASE_PACKAGE ]; then
+        VERSION="$VERSION_SHORT"
+        RELEASE="release"
+    else
+        VERSION="$VERSION_LONG"
+        RELEASE="devel.${LAST_COMMIT_TIMESTAMP}"
+    fi
+}
+
+create_version_file() {
+    echo "--- Creating version file"
+    if [ -f "$VERSION_FILE" ] && [ ! $FORCE ]; then
+        echo "---   Version file already exists (use --force to overwrite)"
+    else
```

Comment:
Overwriting version file **really** sounds like a good idea. We want the version file to be always consistent with actual version of sources. Same about entire metadata directory content.